CHANGELOG.md

```diff
@@ -28,6 +28,11 @@ recommendations of [keepachangelog.com](http://keepachangelog.com/).
 ### Deprecated
 - None
 
+## v0.3.6
+
+### Fixed
+- `Response` instance now includes required `min_api_version` in the root of request params taken from inputs.
```

Comment:
Official docs state that it's [optional](https://developers.viber.com/docs/api/rest-bot-api/#general-send-message-parameters).

Reply:
Unfortunately after using of their API I was not able to use Input 'share-number' without strictly setting the min_api_version in the root of params. The button is not rendering in the app. Another problem that I also found with this gem that the response with error details about failed requests are not showing in the logs 😢

So to figure it out I had to modify the gem source code to get the information about failed request. 

Also I have found a couple of opened issues with this API on another platforms with the same problem of required min api version in case of using inputs that require it.

Reply:
Just now noticed that official API docs don't have `min_api_version` as a parameter on a button level. This makes this solution library specific, as opposed to API specific.

What is preventing the user from doing the same logic on the application level? e.g.
```ruby
locate = Viberroo::Input.reply_button({
  Columns: 4,
  Rows: 3,
  ActionBody: '/near_me',
  min_api_version: 5
}

  browse = Viberroo::Input.url_button({
  Columns: 4,
  Rows: 2,
  ActionBody: 'parrot.live',
  Text: 'Browse something wierd',
  min_api_version: 2
}

rich_message = Viberroo::Message.rich(
  min_api_version: 2,
  rich_media: { 
    ButtonsGroupColumns: 4,
    ButtonsGroupRows: 6,
    Buttons: [search, locate, browse] 
  }
)

def max_api_version(message)
  buttons_array = keyboard.dig(:keyboard, :Buttons)

  return {} unless buttons_array

  buttons_array.map { |btn| btn[:min_api_version] }.compact.max
end

# this can be wrapper in some sort of decorator or something to be nicer
message = rich_message.merge(min_api_version: max_api_version(rich_message))

@bot.send_message(rich_message)
```

Would that solve your problem?

Reply:
> Would that solve your problem?
Yes, it will, but in this case, this will be a custom solution that will do the part of logic that gem should. In this case, I would rather write my own client to work with a Viber API - that's why I decided to contribute here

Reply:
> Can you elaborate? Are you using Rails or something else? Do you expect API errors to be logged?

Yes, I'm using Rails. 
Yes, I expect that API error will be logged

Here is the current behavior:
![image](https://github.com/vikdotdev/viberroo/assets/3767567/aa682017-4d8a-400e-99fe-8526edd2dd21)

And this is what actually goes under the hood:
![image](https://github.com/vikdotdev/viberroo/assets/3767567/7e7f591b-c3ff-4370-910e-91b5fd691bc4)

![image](https://github.com/vikdotdev/viberroo/assets/3767567/e4bf0be9-e0bc-46aa-a606-a440959d6e82)

Reply:
@vikdotdev I think it would be great if this gem could process these codes: [Viber error codes](https://developers.viber.com/docs/api/rest-bot-api/#error-codes)

This will also could solve the issue of missed errors in logs.

Reply:
Alright, that's a lot. Please create a separate issue for logging

Reply:
> Yes, it will, but in this case, this will be a custom solution that will do the part of logic that gem should. In this case, I would rather write my own client to work with a Viber API - that's why I decided to contribute here

Fair enough.